proof/refine/ARM/SchedContextInv_R.thy

```diff
@@ -371,18 +374,393 @@ lemma decode_sc_ctrl_inv_corres:
              split: cap.splits)
   done
 
-(* FIXME RT: preconditions can be reduced, this is what is available at the call site: *)
-lemma invoke_sched_context_corres:
+lemma schedContextBindNtfn_corres:
+  "corres dc
+     (valid_objs and sc_ntfn_sc_at ((=) None) scp
+      and (obj_at (\<lambda>ko. \<exists>ntfn. ko = Notification ntfn \<and> ntfn_sc ntfn = None) ntfnp))
+     (ntfn_at' ntfnp and sc_at' scp)
+     (sched_context_bind_ntfn scp ntfnp) (schedContextBindNtfn scp ntfnp)"
+  unfolding sched_context_bind_ntfn_def schedContextBindNtfn_def
+  apply (clarsimp simp: update_sk_obj_ref_def bind_assoc)
+  apply (rule corres_guard_imp)
+    apply (rule corres_split[OF getNotification_corres])
+      apply (rule corres_guard_imp)
+        apply (rule corres_split[OF setNotification_corres])
+           apply (clarsimp simp: ntfn_relation_def split: Structures_A.ntfn.splits)
+          apply (fold updateSchedContext_def)
```

Comment:
I guess it's not worth changing `schedContextBindNtfn` to actually use `updateSchedContext`?